discretize/CylMesh.py

```diff
@@ -263,111 +874,234 @@ def nodalLaplacian(self):
     @property
     def edgeCurl(self):
         """The edgeCurl property."""
-        if self.nCy > 1:
-            raise NotImplementedError('Edge curl not yet implemented for '
-                                      'nCy > 1')
         if getattr(self, '_edgeCurl', None) is None:
-            # 1D Difference matricies
-            dr = sp.spdiags((np.ones((self.nCx+1, 1))*[-1, 1]).T, [-1, 0],
-                            self.nCx, self.nCx, format="csr")
-            dz = sp.spdiags((np.ones((self.nCz+1, 1))*[-1, 1]).T, [0, 1],
-                            self.nCz, self.nCz+1, format="csr")
-
-            # 2D Difference matricies
-            Dr = sp.kron(sp.identity(self.nNz), dr)
-            Dz = -sp.kron(dz, sp.identity(self.nCx))
-
             A = self.area
             E = self.edge
-            # Edge curl operator
-            self._edgeCurl = (utils.sdiag(1/A)*sp.vstack((Dz, Dr)) *
-                              utils.sdiag(E))
+
+            if self.isSymmetric is True:
```

Comment:
again, I would like to use deflation matrices for this, but it is a bit out of scope for this pr